fl4health/preprocessing/autoencoders/vae_training.py

```diff
@@ -0,0 +1,51 @@
+from typing import Tuple
+
+import numpy as np
+import torch
+
+
+class AETransformer:
+    """Transform function to replace the target in the dataset with data for
+    self-supervised approaches like VAEs. For a conditional model, it can concatenate
+    the condition with the data sample.
+    """
+
+    def __init__(self, condition: str = "", img_dims: int = 1) -> None:
+        """Initializes the AETransformer with an optional condition.
+
+        Args:
+            condition (str, optional): Condition for the transformation. Defaults to "".
+        """
+        self.condition = condition
+        self.img_dims = img_dims
+
+    def __call__(self, sample: np.ndarray, target: np.ndarray) -> Tuple[torch.Tensor, torch.Tensor]:
+        """Applies the transformation to the input data.
+
+        Args:
+            sample (np.ndarray): Data sample from the dataset.
+            target (Optional[np.ndarray], optional): Target is only necessary
+            in CVAEs conditioned on data label. Defaults to None.
+
+        Returns:
+            Tuple[torch.Tensor, torch.Tensor]: The data format used for training,
+            for self-supervised VAE it is (sample, sample). For CVAE conditioned on
+            an integer it is ([sample, int], sample). For CVAE conditioned on label,
+            it is ([sample, target], sample)
```

Comment:
We might want to be more general than this. That is, CVAEs can be conditioned on any kind of torch.Tensor. So I'd say the transformer here should take a `conditional_tensor` rather than a "target" tensor and if the tensor passed to the transformer happens to be a label that is a "happy coincidence"